action.yml

```diff
@@ -317,22 +336,83 @@ runs:
+            if [[ $TRIGGER_CI_EXCEEDED == "" ]]; then
+              TRIGGER_CI_EXCEEDED=$INTO_BRANCH
+            else
+              TRIGGER_CI_EXCEEDED="$TRIGGER_CI_EXCEEDED,$INTO_BRANCH"
+            fi
           fi
         done
+        echo "TRIGGER_CI_BRANCH_01 is $TRIGGER_CI_BRANCH_01"
+        echo "TRIGGER_CI_BRANCH_02 is $TRIGGER_CI_BRANCH_02"
+        echo "TRIGGER_CI_BRANCH_03 is $TRIGGER_CI_BRANCH_03"
+        echo "TRIGGER_CI_BRANCH_04 is $TRIGGER_CI_BRANCH_04"
+        echo "TRIGGER_CI_BRANCH_05 is $TRIGGER_CI_BRANCH_05"
+        echo "TRIGGER_CI_BRANCH_06 is $TRIGGER_CI_BRANCH_06"
+        echo "TRIGGER_CI_EXCEEDED is $TRIGGER_CI_EXCEEDED"
+        echo "trigger_ci_branch_01=$TRIGGER_CI_BRANCH_01" >> $GITHUB_OUTPUT
+        echo "trigger_ci_branch_02=$TRIGGER_CI_BRANCH_02" >> $GITHUB_OUTPUT
+        echo "trigger_ci_branch_03=$TRIGGER_CI_BRANCH_03" >> $GITHUB_OUTPUT
+        echo "trigger_ci_branch_04=$TRIGGER_CI_BRANCH_04" >> $GITHUB_OUTPUT
+        echo "trigger_ci_branch_05=$TRIGGER_CI_BRANCH_05" >> $GITHUB_OUTPUT
+        echo "trigger_ci_branch_06=$TRIGGER_CI_BRANCH_06" >> $GITHUB_OUTPUT
+        echo "trigger_ci_exceeded=$TRIGGER_CI_EXCEEDED" >> $GITHUB_OUTPUT
+
+    - name: Trigger CI 01
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_01 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_01 }}
+
+    - name: Trigger CI 02
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_02 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_02 }}
+
+    - name: Trigger CI 03
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_03 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_03 }}
+
+    - name: Trigger CI 04
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_04 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_04 }}
+
+    - name: Trigger CI 05
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_05 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_05 }}
+
+    - name: Trigger CI 06
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_branch_06 != '' }}
+      uses: silverstripe/gha-trigger-ci@v1
+      with:
+        branch: ${{ steps.git-merge-up.outputs.trigger_ci_branch_06 }}
+    
+    - name: Trigger CI Exceeded
+      if: ${{ steps.git-merge-up.outputs.trigger_ci_exceeded != '' }}
+      shell: bash
+      env:
+        TRIGGER_CI_EXCEEDED: ${{ steps.git-merge-up.outputs.trigger_ci_exceeded }}
+      run: |
+        echo "Exceeded the number of branches that can be have CI triggered"
+        echo "Branches where CI was not triggered is $TRIGGER_CI_EXCEEDED"
+        exit 1
```

Comment:
Why have this as a separate step? We already throw an exception early if there's >6 branches, so if we're in this scenario we should exit out because it means we should have _already_ failed the job, but the exception didn't get thrown when it should have.

Reply:
It's just an extra check in case we increase the 6 branch limit in the future in PHP though forget to do it in yml

Reply:
I'd prefer to not do this as an extra step, because having a whole step just to say "we did a booboo" is really weird. But it's not worth holding off merging just for that.